lib/bamboo/mailer.ex

```diff
@@ -64,17 +64,14 @@ defmodule Bamboo.Mailer do
     quote bind_quoted: [opts: opts] do
       config = Bamboo.Mailer.parse_opts(__MODULE__, opts)
```

Comment:
`parse_opts` was doing a `Keyword.fetch!(opts, :otp_app)` to enforce that it wasn't omitted. Going to include that in the macro so that can get caught during compile time.